numpy/lib/function_base.py

```diff
@@ -989,7 +989,17 @@ def average(a, axis=None, weights=None, returned=False):
     TypeError: Axis must be specified when shapes of a and weights differ.
 
     """
-    a = np.asanyarray(a)
+    # replace the next few lines with "a = np.asanyarray(a)" in the future
```

Comment:
If doing this, should add comment with data and release to help track when to change. Something like
```
# 3/19/2016 1.12.0
```